### PR TITLE
FTM : Avoid FPE when FT_MOTION is motion system

### PR DESCRIPTION
**Pull request by narno2202**
### Description
In the last bugfix, using MarlinSimulator to test a menu, when FT_MOTION is active, MarlinSimulator crashes with a floating point exception. It's totally independent of the trajectory type, axes synchronisation, smoothing, and linear advance. There is a division by zero in `stepping.cpp` when `stepper_plan.advance_dividend_q0_32` for an axis is zero.

Add a test in `#define _DIVIDEND_RECIP(A)` macro to avoid division by zero.

### Requirements
`FT_MOTION` enabled and active

### Benefits
Avoid division by zero and potential crash

### CODE REVIEW

**Comment by dbuezas** [comment]
The reciprocal is not used when advance is zero, so i left the test out to save a branch. I guess the simulator is more strict?

**Comment by dbuezas** [comment]
```cpp

#define RUN_AXIS(A) if(advance_q32.A > 0) NOMORE(iterations, uint32_t((uint64_t(space_q32.A) * advance_dividend_reciprocal.A) >> 32));

```

**Comment by dbuezas** [comment]
Maybe it would be best to keep your fix but set the reciprocal to UINT32_MAX so the if in the define above can be removed. That one runs a lot more often so it would ve a bit more perf

**Comment by narno2202** [comment]
The error in only in the simulator so it's probably platform related. PR updated.

**Comment by dbuezas** [comment]
This is a net win, 👍

File: Marlin/src/module/ft_motion/stepping.cpp [hunk 1]
```diff
@@ -101,7 +101,7 @@ uint32_t Stepping::plan() {
   // The reciprocal is actually 2^32/dividend, but that requires dividing a uint64_t, which quite expensive
   // Since even the real reciprocal may underestimate the quotient by 1 anyway already, this optimisation doesn't
   // make things worse. This underestimation is compensated for in advance_until_step.
-  #define _DIVIDEND_RECIP(A) advance_dividend_reciprocal.A = UINT32_MAX / stepper_plan.advance_dividend_q0_32.A;
+  #define _DIVIDEND_RECIP(A) (stepper_plan.advance_dividend_q0_32.A ==0) ? advance_dividend_reciprocal.A = UINT32_MAX : advance_dividend_reciprocal.A = UINT32_MAX / stepper_plan.advance_dividend_q0_32.A;
```
Comment (dbuezas) [inline Marlin/src/module/ft_motion/stepping.cpp]:
```suggestion
  #define _DIVIDEND_RECIP(A) advance_dividend_reciprocal.A = (stepper_plan.advance_dividend_q0_32.A == 0) ? UINT32_MAX : UINT32_MAX / stepper_plan.advance_dividend_q0_32.A;
```

Comment (dbuezas) [reply Marlin/src/module/ft_motion/stepping.cpp]: Or 

```suggestion
  #define _DIVIDEND_RECIP(A) advance_dividend_reciprocal.A = UINT32_MAX / (stepper_plan.advance_dividend_q0_32.A == 0 ? 1 : stepper_plan.advance_dividend_q0_32.A);
```

Both the same as yours but shorter.

Although divide by one may be slower than not doing it, up to you.

**Comment by narno2202** [comment]
I prefer the shorter one. Going back to resonance testing in FT_MOTION
